systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -425,16 +443,22 @@ void testMirrorMaker2TlsAndScramSha512Auth() {
             .build();
         resourceManager.createResourceWithWait(targetClients.consumerScramShaTlsStrimzi(testStorage.getTargetClusterName()));
         ClientUtils.waitForInstantConsumerClientSuccess(testStorage);
-        LOGGER.info("Messages successfully mirrored");
 
-        // TODO https://github.com/strimzi/strimzi-kafka-operator/issues/8864
-        /*if (!Environment.isUnidirectionalTopicOperatorEnabled()) {
-            KafkaTopicUtils.waitForKafkaTopicCreation(testStorage.getNamespaceName(), testStorage.getMirroredSourceTopicName());
-            KafkaTopic mirroredTopic = KafkaTopicResource.kafkaTopicClient().inNamespace(testStorage.getNamespaceName()).withName(testStorage.getMirroredSourceTopicName()).get();
+        LOGGER.info("Checking topic is mirrored correctly in target cluster");
 
-            assertThat(mirroredTopic.getSpec().getPartitions(), is(3));
-            assertThat(mirroredTopic.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL), is(testStorage.getTargetClusterName()));
-        }*/
+        // deploy admin client
+        resourceManager.createResourceWithWait(
+            AdminClientTemplates.scramShaOverTlsAdminClient(
+                testStorage.getNamespaceName(),
+                testStorage.getTargetUsername(),
+                testStorage.getAdminName(),
+                testStorage.getTargetClusterName(),
+                KafkaResources.tlsBootstrapAddress(testStorage.getTargetClusterName())
+        ));
+        final AdminClient targetClusterAdminClient = AdminClientUtils.getConfiguredAdminClient(testStorage.getNamespaceName(), testStorage.getAdminName());
+
+        AdminClientUtils.waitForTopicPresence(targetClusterAdminClient, testStorage.getMirroredSourceTopicName());
+        assertThat(targetClusterAdminClient.describeTopic(testStorage.getMirroredSourceTopicName()).partitionCount(), is(3));
```

Comment:
Wouldn't be better to get such value via fabri8? I mean that magic `3`. Or a better way is to factor out that 3 with local variable :) also in creating i.e.,
```diff
- KafkaTopicTemplates.topic(testStorage.getSourceClusterName(), testStorage.getTopicName(), 3, testStorage.getNamespaceName()).build(),
+ KafkaTopicTemplates.topic(testStorage.getSourceClusterName(), testStorage.getTopicName(), sourceTopicPartitions, testStorage.getNamespaceName()).build(),
 ```